polarion/export_to_sheets.py

```diff
@@ -0,0 +1,111 @@
+from tracemalloc import start
+from gspread_formatting import set_column_width
+import datetime
+
+
+class export_data:
+    import gspread
+
+    from gspread.exceptions import SpreadsheetNotFound
+
+    def next_available_row(self, worksheet):
+        """
+        Function calculates next available row in the gsheet
+        """
+        str_list = list(filter(None, worksheet.col_values(1)))
+        return str(len(str_list) + 1)
+
+    def format_cell(self):
+        return {
+            "horizontalAlignment": "CENTER",
+            "borders": {
+                "top": {"style": "SOLID", "width": 1},
+                "bottom": {"style": "SOLID", "width": 1},
+                "left": {"style": "SOLID", "width": 1},
+                "right": {"style": "SOLID", "width": 1},
+            },
+        }
+
+    """def get_last_day(self,ws,last_row):
+        day=ws.acell('A'+str(last_row)).value
+        day=day.split("Day")
+        if day[1]=='':
+            return "Day1"
+        else:
+            return "Day"+str(int(day[1])+1)"""
+
+    def export_sheets(self, data, config):
+        """
+        Function to export the data fetched from polarion to gsheet.
+        """
+        # gc = self.gspread.oauth(authorized_user_filename=config.GS_CREDENTIALS)
+        gc = self.gspread.service_account(filename=config.GS_CREDENTIALS)
+        try:
+            sh = gc.open(config.file_name)
+        except self.SpreadsheetNotFound:
+            print(
+                "Spreadsheets in not exits or you have not shared with client email id"
+            )
+        ws = sh.get_worksheet(0)
+        if self.next_available_row(ws) == "1":
+            # sh = gc.create(config.file_name)
+            # ws = sh.get_worksheet(0)
+            ws.update("A1", "Tier Classification")
+            set_column_width(ws, "A", 150)
+
+            start_c = "B"
+            for keys in data.keys():
+                end_c = chr(ord(start_c) + 2)
+                cell_merge_name = start_c + str(1) + ":" + end_c + str(1)
+                ws.merge_cells(cell_merge_name)
+                print(start_c + str(1))
+                ws.update(start_c + str(1), keys.capitalize())
+                t_s = start_c
+                """for val in self.list_columns:
+                    ws.update(t_s+str(3),val)
+                    t_s=chr(ord(t_s)+1)
+                """
+                for filter_keys in config.filter.keys():
+                    for val, label in zip(
+                        config.filter[filter_keys]["values"],
+                        config.filter[filter_keys]["labels"],
+                    ):
+                        query = (
+                            r"https://polarion.engineering.redhat.com/polarion/#/workitems?query="
```

Comment:
Can we gather the endpoint URL from the config file

Reply:
The only change needed here is project.id -> which is kept in config file and since queries are not dynamic, this is kept in the code itself. The query becomes generic based on the "project.id"